design-proposals/volume-claim-templates.md

```diff
@@ -0,0 +1,181 @@
+# Overview
+This proposal is about having a better integration with kubernetes storage APIs.  
+With recent support of kubernetes [volume populators](https://kubernetes.io/blog/2022/05/16/volume-populators-beta/) we have implemented [CDI volume populators](https://github.com/kubevirt/containerized-data-importer/blob/main/doc/cdi-populators.md) which allows population of PVC the kubernetes way without the need for DataVolumes.  
+Along major benefits of using this populators we loss one of DataVolumes greatest achievements - generating a PVC with the optimal properties fitting the user's storage according to [Storage Profile](https://github.com/kubevirt/containerized-data-importer/blob/main/doc/storageprofile.md).  
+This design should help us gain the best of all worlds.
+
+## Motivation
+DataVolumes were our only way of having a populated PVC with a VM disk image. While solving this crucial problem for VM's in kubernetes, Datavolumes certainly have downsides, mostly with regard to integrating with backup/Disaster Recovery solutions. Now that kubernetes has a general solution to the specific problem that DataVolumes solve we can prevent from using Datavolumes.  
+Harnessing k8s and CDI new volume populators API along with our knowledge of storage capabilities will allow the user to enjoy synchronization with k8s and better compatibility with GitOps, Backup and Restore, and Disaster Recovery workflows as well as the smart creation of PVCs that best suits the user's storage.  
+
+## Goals
+- Provide a method for creating VMs with PVCs that suits best the user's storage while using CDI volume populators.
+
```

Comment:
Not a goal, but a requirement: Must work with VMs created with `VM.metadata.generateName`